src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,210 @@
+package duke.command;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.dukeexception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+public class Command {
+    /** The input command by user */
+    private String commandName;
+
+
+    /**
+     * Class constructor.
+     */
+    public Command(String commandName) {
+        this.commandName = commandName;
+    }
+
+
+    /**
+     * Processes tasks according to the command.
+     *
+     * @param tasks  the tasks before execution.
+     * @throws IOException  If an input or output
+     *                      exception occurred
+     */
+    public void execute(TaskList tasks) throws IOException, DukeException {
```

Comment:
This method seems a bit too long. Perhaps you could have divided it into smaller methods?